test/e2e/go.mod

```diff
@@ -163,18 +165,18 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	k8s.io/apiextensions-apiserver v0.27.2 // indirect
 	k8s.io/apiserver v0.27.2 // indirect
-	k8s.io/cli-runtime v0.27.2 // indirect
+	k8s.io/cli-runtime v0.28.0 // indirect
```

Comment:
I still bumped k8s.io/kubectl to 0.28.0 and several other indirect dependencies are updated automatically. Otherwise `go test e2e_suite_test.go` will have compile error:

```
go test e2e_suite_test.go
# k8s.io/cli-runtime/pkg/resource
../../../../../../pkg/mod/k8s.io/cli-runtime@v0.27.2/pkg/resource/query_param_verifier.go:83:38: cannot use oapi (variable of type *"github.com/google/gnostic-models/openapiv2".Document) as *"github.com/google/gnostic/openapiv2".Document value in argument to supportsQueryParam
../../../../../../pkg/mod/k8s.io/cli-runtime@v0.27.2/pkg/resource/query_param_verifier.go:86:36: cannot use oapi (variable of type *"github.com/google/gnostic-models/openapiv2".Document) as *"github.com/google/gnostic/openapiv2".Document value in argument to supportsQueryParam
# k8s.io/kubectl/pkg/util/openapi
../../../../../../pkg/mod/k8s.io/kubectl@v0.27.2/pkg/util/openapi/openapi_getter.go:36:42: cannot use &CachedOpenAPIGetter{} (value of type *CachedOpenAPIGetter) as discovery.OpenAPISchemaInterface value in variable declaration: *CachedOpenAPIGetter does not implement discovery.OpenAPISchemaInterface (wrong type for method OpenAPISchema)
		have OpenAPISchema() (*"github.com/google/gnostic/openapiv2".Document, error)
		want OpenAPISchema() (*"github.com/google/gnostic-models/openapiv2".Document, error)
../../../../../../pkg/mod/k8s.io/kubectl@v0.27.2/pkg/util/openapi/openapi_getter.go:49:28: cannot use g.openAPIClient.OpenAPISchema() (value of type *"github.com/google/gnostic-models/openapiv2".Document) as *"github.com/google/gnostic/openapiv2".Document value in assignment
../../../../../../pkg/mod/k8s.io/kubectl@v0.27.2/pkg/util/openapi/openapi_getter.go:78:46: cannot use oapi (variable of type *"github.com/google/gnostic-models/openapiv2".Document) as *"github.com/google/gnostic/openapiv2".Document value in argument to NewOpenAPIData
../../../../../../pkg/mod/k8s.io/kubectl@v0.27.2/pkg/util/openapi/openapi.go:52:38: cannot use doc (variable of type *"github.com/google/gnostic/openapiv2".Document) as *"github.com/google/gnostic-models/openapiv2".Document value in argument to proto.NewOpenAPIData
FAIL	command-line-arguments [build failed]
```


I didn't bump k8s.io/cluster-bootstrap or sigs.k8s.io/controller-runtime because bump either will throw other  incompatible compile errors. 
We can bump capi to 1.6 and sigs.k8s.io/controller-runtime to 0.16, along with k8s.io/cluster-bootstrap in future.

Reply:
We can probably bump to a version of CAPI main once the CR bump PR there merges (probably today). Just like we used a v1.5 pre-release version (l.185-188)